nearcache/nearcache_it_test.go

```diff
@@ -955,10 +969,114 @@ func TestMemberSetAllInvalidatesClientNearCache(t *testing.T) {
         	map.setAll(items);
 		`, size, tcx.MapName)
 	}
-	memberInvalidatesClientNearCache(t, f)
+	memberInvalidatesClientNearCache(t, 51021, f)
+}
+
+func TestForceRepairingTaskRun(t *testing.T) {
+	// TODO: mark this test slow.
+	// this is a test for covering async init of RepairingTask.
+	clusterName := t.Name()
+	mapName := it.NewUniqueObjectName("map")
+	const port = 53001
+	clsCfg := invalidationXMLConfig(clusterName, "non-existent", port)
+	cls := it.StartNewClusterWithConfig(1, clsCfg, port)
+	const mapSize = 1000
+	// populate server side map
+	for i := 0; i < mapSize; i++ {
+		v := strconv.Itoa(i)
+		it.MapSetOnServer(cls.ClusterID, mapName, v, v)
+	}
+	// add client with Near Cache
+	ctx := context.Background()
+	cfg := cls.DefaultConfig()
+	cfg.NearCacheInvalidation.SetReconciliationIntervalSeconds(30)
+	cfg.NearCacheInvalidation.SetMaxToleratedMissCount(1)
+	ncc := nearcache.Config{
+		Name: mapName,
+	}
+	// setting invalidation on change to false, since we'll handle that in the test.
+	ncc.SetInvalidateOnChange(true)
+	cfg.AddNearCache(ncc)
+	client := it.MustClient(hz.StartNewClientWithConfig(nil, cfg))
+	defer client.Shutdown(ctx)
+	ic := hz.NewClientInternal(client)
+	mgr := ic.NewNearCacheManager(1, 1)
+	defer mgr.Stop()
+	m := it.MustValue(client.GetMap(ctx, mapName)).(*hz.Map)
+	nca := hz.MakeNearCacheAdapterFromMap(m).(it.NearCacheAdapter)
+	for i := int32(0); i < mapSize; i++ {
+		if _, err := m.Get(ctx, i); err != nil {
+			t.Fatal(err)
+		}
+	}
+	cls.Shutdown()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		time.Sleep(2 * time.Second)
+		cls = it.StartNewClusterWithConfig(1, clsCfg, port)
+		wg.Done()
+	}()
+	defer cls.Shutdown()
```

Comment:
I think this defer call targets the previous cluster, can you verify if the latest cluster is being shutdown?

Reply:
Printing the address of the `cls` variable at: https://github.com/hazelcast/hazelcast-go-client/pull/850/commits/f362a6d95526a3bd7abd9778b3d0ec6fafc44868